### PR TITLE
rename ControllerPreset to LegacyControllerMapping

### PR DESCRIPTION
**Pull request by Be-ing**
This is the last step before decoupling controller hardware I/O from the legacy mapping system.

**This is strictly a renaming PR. IF YOU ASK ME TO EXPAND THE SCOPE OF THIS, INCLUDING CODE CLEANUP, I WILL CLOSE THIS PR AND STOP WORKING ON THIS.**

### CODE REVIEW

File: src/controllers/controllermappinginfoenumerator.cpp [hunk 1]
```diff
@@ -23,32 +23,32 @@ bool presetInfoNameComparator(const PresetInfo &a, const PresetInfo &b) {
...
-PresetInfoEnumerator::PresetInfoEnumerator(const QStringList& searchPaths)
+MappingInfoEnumerator::MappingInfoEnumerator(const QStringList& searchPaths)
         : m_controllerDirPaths(searchPaths) {
-    loadSupportedPresets();
+    loadSupportedMappings();
 }
 
-QList<PresetInfo> PresetInfoEnumerator::getPresetsByExtension(const QString& extension) {
+QList<MappingInfo> MappingInfoEnumerator::getMappingsByExtension(const QString& extension) {
     if (extension == MIDI_PRESET_EXTENSION) {
```
Comment (uklotzde) [inline src/controllers/controllermappinginfoenumerator.cpp:36]: Do we want to keep the `_PRESET_EXTENSION` names or rename them to `_MAPPING_EXTENSION` for consistency

File: src/controllers/hid/hidcontroller.cpp [hunk 2]
```diff
@@ -34,24 +34,24 @@ HidController::~HidController() {
     }
 }
 
-QString HidController::presetExtension() {
+QString HidController::mappingExtension() {
     return HID_PRESET_EXTENSION;
```
Comment (uklotzde) [inline src/controllers/hid/hidcontroller.cpp:38]: Ok, this usage suggests to rename the \#defines as well.

**Comment by uklotzde** [comment]
Just one minor comment about consistency.

I have checked that the group name in the preferences remains unchanged for backwards compatibility. It would be helpful to mention these exceptions in the PR description explicitly.

**Comment by uklotzde** [comment]
It might be helpful leave a note about the (now) inconsistent naming in the code where the key "[ControllerPreset]" is defined.

**Comment by Be-ing** [comment]
Notes addressed, CI passed.

File: src/controllers/controllermanager.cpp [hunk 3]
```diff
@@ -218,7 +218,7 @@ QList<Controller*> ControllerManager::getControllerList(bool bOutputDevices, boo
     return filteredDeviceList;
 }
 
-QString ControllerManager::getConfiguredPresetFileForDevice(const QString& name) {
+QString ControllerManager::getConfiguredMappingFileForDevice(const QString& name) {
     return m_pConfig->getValueString(ConfigKey("[ControllerPreset]", sanitizeDeviceName(name)));
```
Comment (uklotzde) [inline src/controllers/controllermanager.cpp:222]: Unfortunately, this string is duplicated. I suggest to extract into into a local constant and move the comment there.

**Review by uklotzde — APPROVED** [review]
Confirmed, just a renaming to avoid confusion between deprecated and new components. LGTM

**Pull request merged**